storage/src/main/java/zipkin2/storage/kafka/KafkaStoreHttpService.java

```diff
@@ -0,0 +1,365 @@
+import static zipkin2.storage.kafka.streams.TraceStoreTopologySupplier.SERVICE_NAMES_STORE_NAME;
+import static zipkin2.storage.kafka.streams.TraceStoreTopologySupplier.SPAN_IDS_BY_TS_STORE_NAME;
+import static zipkin2.storage.kafka.streams.TraceStoreTopologySupplier.SPAN_NAMES_STORE_NAME;
+import static zipkin2.storage.kafka.streams.TraceStoreTopologySupplier.TRACES_STORE_NAME;
+
+/**
+ * Server to enable access to local stores.
+ * <p>
+ * Given the partitioned nature of local stores, a RPC layer is required to allow accessing
+ * distributed state. This component exposes access to local state via Http call from {@link
+ * KafkaSpanStore}
+ *
+ * @since 0.6.0
+ */
+public class KafkaStoreHttpService implements Consumer<ServerBuilder> {
+  static final Logger LOG = LoggerFactory.getLogger(KafkaStoreHttpService.class);
+  static final ObjectMapper MAPPER = new ObjectMapper();
+
+  final KafkaStorage storage;
+  final long minTracesStored;
+
+  KafkaStoreHttpService(KafkaStorage storage) {
+    this.storage = storage;
+    this.minTracesStored = storage.minTracesStored;
+  }
+
+  @Override public void accept(ServerBuilder serverBuilder) {
+    serverBuilder.annotatedService("/zipkin/storage/kafka", this);
+  }
+
+  @Get("/dependencies")
+  public AggregatedHttpResponse getDependencies(
+      @Param("endTs") long endTs,
+      @Param("lookback") long lookback) {
+    try {
+      ReadOnlyWindowStore<Long, DependencyLink> dependenciesStore =
+          storage.getDependencyStoreStream().store(DEPENDENCIES_STORE_NAME,
+              QueryableStoreTypes.windowStore());
+      List<DependencyLink> links = new ArrayList<>();
+      Instant from = Instant.ofEpochMilli(endTs - lookback);
+      Instant to = Instant.ofEpochMilli(endTs);
+      dependenciesStore.fetchAll(from, to)
+          .forEachRemaining(keyValue -> links.add(keyValue.value));
+      List<DependencyLink> mergedLinks = DependencyLinker.merge(links);
+      LOG.debug("Dependencies found from={}-to={}: {}", from, to, mergedLinks.size());
+      return AggregatedHttpResponse.of(
+          HttpStatus.OK,
+          MediaType.JSON,
+          DependencyLinkBytesEncoder.JSON_V1.encodeList(mergedLinks));
+    } catch (InvalidStateStoreException e) {
+      LOG.warn("State store is not ready", e);
+      return AggregatedHttpResponse.of(
+          HttpStatus.OK,
+          MediaType.JSON,
+          DependencyLinkBytesEncoder.JSON_V1.encodeList(Collections.emptyList()));
+    }
+  }
+
+  @Get("/serviceNames")
+  public AggregatedHttpResponse getServiceNames() {
+    try {
+      ReadOnlyKeyValueStore<String, String> store =
+          storage.getTraceStoreStream()
+              .store(SERVICE_NAMES_STORE_NAME, QueryableStoreTypes.keyValueStore());
+      //Set<String> names = new HashSet<>();
+      ArrayNode array = MAPPER.createArrayNode();
+      store.all().forEachRemaining(keyValue -> array.add(keyValue.value));
+      return AggregatedHttpResponse.of(HttpStatus.OK, MediaType.JSON,
```

Comment:
I realized I was being too narrow with my review - this is an annotated service, it provides much more power than just path mapping so let's use it ;) 

You can return `JsonNode` (`ArrayNode` is a subclass) directly and Armeria will automatically serialize it and return a response with the JSON content type.

If instead you want to return an object, you can add `@ProducesJson` annotation and return the object.

https://line.github.io/armeria/server-annotated-service.html

Reply:
Sorry for realizing after I asked you to add ObjectMapper all over the place here ><

Reply:
np :) apply some changes now